arroyo/processing/strategies/produce.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+from collections import deque
+from concurrent.futures import Future
+from typing import Deque, Optional, Tuple
+
+from arroyo.backends.abstract import Producer
+from arroyo.processing.strategies.abstract import MessageRejected, ProcessingStrategy
+from arroyo.types import Commit, Message, Position, Topic, TPayload
+
+logger = logging.getLogger(__name__)
+
+
+class ProduceAndCommit(ProcessingStrategy[TPayload]):
+    """
+    For each message received, produces a message to the topic being passed in.
+    Waits on the futures and only once completed, commits the offset of the message.
```

Comment:
Please let's elaborate on this.
The API documentation will mostly be generated from docstring (or will be the docstring).
I would specifically cover the delivery guarantee (at least once), the error management, and what to do with `MessageRejected` exceptions.

Reply:
Updated